### ISSUE TITLE
Automatic re-opening of terminal windows when loading a project

### ISSUE THREAD

**Issue by gerhardvr-finclude** [microsoft/vscode#197779]
I woud like VSCode to automatically re-open all the terminal windows that was opened at the time a project/VSCode was closed. For example, if I have a Git Bash terminal, a Powershell terminal and a Javascript debug terminal open at the time I close my project, I'd like all of these terminal to automatically open again next time I open the project.

I currently have "terminal.integrated.enablePersistentSessions": true set in my VSCode settings file but this opens only one of the terminals automatically, even though there were multiple terminal open at the time the project was closed. In addition, there seems to be logic to which one of the terminal windows it will open, and soemtimes it will open the Powershell terminal, even though it wasn't even open when the project was closed.

**Comment by meganrogge** [comment]
This should work - we only persist terminals that have had input. A gif would help. 

Also, what do you have `terminal.integrated.persistentSessionReviveProcess` set to?

**Comment by gerhardvr-finclude** [comment]
You are correct, it does indeed open multiple terminal windows as long as they had some input. I was not aware that they would only be re-opened when they had input.

However, I tested with 4 open terminal windows and only three of the four were re-opened. For some reason the Javascript Debug Terminal is not re-opened, even if I were debugging using it, before close.

Before close
[cid:1599c806-2123-4fc1-9eab-2313c71459df]

After re-open
[cid:8735b863-157f-4dc5-b2cd-6ce427d48c70]



Best regards



Gerhard van Rensburg


________________________________

**Comment by gerhardvr-finclude** [comment]
You are correct, it does indeed open multiple terminal windows as long as they had some input. I was not aware that they would only be re-opened when they had input.

However, I tested with 4 open terminal windows and only three of the four were re-opened. For some reason the Javascript Debug Terminal is not re-opened, even if I were debugging using it, before close.

![before close](https://github.com/microsoft/vscode/assets/78418254/8857e741-3f35-450e-98bf-6308d348b4cf)

![After open](https://github.com/microsoft/vscode/assets/78418254/79a29341-efb7-480e-85ec-beaa38641889)

**Comment by meganrogge** [comment]
That's by design - we don't persist extension created terminals.
